### ISSUE TITLE
R per-package project lib paths

### ISSUE THREAD

**Issue by wwarriner** [uabrc/uabrc.github.io#264]
Copying wholesale from the old docs. This is a good practice if you are working on more than one project (and a good practice in general for any environment management).

https://docs.uabgrid.uab.edu/wiki/R#Per-Project_Package_Libraries

```
Per-Project Package Libraries

When working with multiple projects, or when using software like [AFNI](https://docs.uabgrid.uab.edu/wiki/AFNI) which make use of R internally, it may be helpful to use separate folders to store libraries for separate projects and software. Keeping library paths separate on a per-project or per-software basis will minimize the risk of library conflicts and hard-to-trace bugs.

Library paths may be managed within R using the [libPaths](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/libPaths) function. Simply pass a list of directories to the function to change the library paths available to R.

The following reference assumes that an R module is already loaded. To achieve separation of libraries on a per-project bases, navigate to one of the desired project directory. Create a folder called "rlibs", which will be used to store packages. Create an empty text file called ".Rprofile" in the root project directory, if it doesn't already exist. Add the following to the ".Rprofile". If a call to libPaths already exists, exercise judgement before modifying it.

.libPaths(c('./rlibs'))

When using...

    [RStudio](https://docs.uabgrid.uab.edu/wiki/RStudio): you will need to set your root project folder as the working directory. Run the ".Rprofile" file you created earlier to set the "rlibs" folder as the default library path. All newly installed packages in this session will be installed in that folder. 

    R REPL (read-eval-print loop): started by calling "R" at the command line. The ".Rprofile" file will be loaded and executed as R code by the REPL environment before giving control to you. This will set the "rlibs" folder as the default library path. All newly installed packages in this session will be installed in that folder. 

    Rscript: ensure that you start any scripts from the directory containing the ".Rprofile". As with the R REPL environment, the file will be executed prior to running any other code, setting the "rlibs" folder as the default library path. All newly installed packages in this script will be installed in that folder. 

Repeating this process for each project ensures that libraries are kept separate, allowing more flexibility, repeatability, and reducing the risk of errors or cross-contamination of versions and dependencies. The only downside is maintaining discipline in creating the folder and file each time a new project is started, and the additional maintenance relating to each separate library.

It is possible for collisions to still occur with packages installed in the default locations. If you wish to use the practice described above, you may need to remove packages installed in the default locations.

If you have a single workflow using multiple versions of R that are causing package collisions, please contact us for [Support](https://docs.uabgrid.uab.edu/wiki/Support). We will work with you to find an optimal solution.
```

**Comment by mdefende** [comment]
This is better done with the renv package and R Projects. I can write a section on this and add it to the Workflow Solutions page

**Comment by wwarriner** [comment]
Closed in favor of #290
